File: rb/features.rb
# ConsoleWeatherForecast SDK feature factory

require_relative 'feature/base_feature'
require_relative 'feature/ratelimit_feature'
require_relative 'feature/retry_feature'
require_relative 'feature/test_feature'
require_relative 'feature/timeout_feature'


module ConsoleWeatherForecastFeatures
  def self.make_feature(name)
    case name
    when "base"
      ConsoleWeatherForecastBaseFeature.new
    when "ratelimit"
      ConsoleWeatherForecastRatelimitFeature.new
    when "retry"
      ConsoleWeatherForecastRetryFeature.new
    when "test"
      ConsoleWeatherForecastTestFeature.new
    when "timeout"
      ConsoleWeatherForecastTimeoutFeature.new
    else
      ConsoleWeatherForecastBaseFeature.new
    end
  end
end

File: py/features.py
# ConsoleWeatherForecast SDK feature factory

from feature.base_feature import ConsoleWeatherForecastBaseFeature
from feature.test_feature import ConsoleWeatherForecastTestFeature


def _make_feature(name):
    features = {
        "base": lambda: ConsoleWeatherForecastBaseFeature(),
        "test": lambda: ConsoleWeatherForecastTestFeature(),
    }
    factory = features.get(name)
    if factory is not None:
        return factory()
    return features["base"]()
